### PR TITLE
[wip] Syslog module

### PR DESCRIPTION
**Pull request by blotus**
Basic support for syslog acquisition:
 - [x] syslog server
 - [x] documentation
 - [x] Tests

### CODE REVIEW

File: pkg/acquisition/modules/syslog/syslog.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+	if pid != nil {
+		/*
+			!!! ugly hack !!!
+			Due to a bug in the syslog parser we use (https://github.com/influxdata/go-syslog/issues/31),
+			the ProcID field will contain garbage if the message as a ] anywhere in it.
+			Assume that a correctly formated ProcID only contains number, and if this is not the case, set it to an arbitrary value
+		*/
+		_, err := strconv.Atoi(*pid)
+		if err != nil {
+			ret += "[4242]: "
```
Comment (sabban) [inline pkg/acquisition/modules/syslog/syslog.go:143]: as an arbitrary value I would choose sth rather distinguishable like 0 (is it possible to have 0, or 1 as we all know that 1 is a very low probable value for this ?)

Comment (blotus) [reply pkg/acquisition/modules/syslog/syslog.go:143]: Using 0 is not possible, as the grok pattern we use to parse a syslog line expects a positive integer in the PID field.

File: pkg/acquisition/modules/syslog/syslog.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	for {
+		select {
+		case <-t.Dying():
+			s.logger.Info("Syslog datasource is dying")
+			err := s.server.KillServer()
+			return err
+		case syslogLine := <-c:
+			var line string
+			var ts time.Time
+
```
Comment (buixor) [inline pkg/acquisition/modules/syslog/syslog.go:187]: can we have prom metrics tracking events per source (ie. network client ip addr etc.)

File: pkg/acquisition/modules/syslog/syslog.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+			linesReceived.With(prometheus.Labels{"source": fmt.Sprintf("%s:%d", s.config.Addr, s.config.Port)}).Inc()
+			p := rfc5424.NewParser()
+			m, err := p.Parse(syslogLine.Message)
+			if err != nil {
+				s.logger.Debugf("could not parse message as RFC5424, falling back to RFC3164 : %s", err)
+				p = rfc3164.NewParser(rfc3164.WithYear(rfc3164.CurrentYear{}))
+				m, err = p.Parse(syslogLine.Message)
+				if err != nil {
+					s.logger.Errorf("could not parse message: %s", err)
```
Comment (buixor) [inline pkg/acquisition/modules/syslog/syslog.go:196]: can we .WithField to associate error to network source ? (is true for the rest of the function as well)

**Review by buixor — CHANGES_REQUESTED** [review]
overall lgtm

File: docs/v1.X/docs/user_guide/data_sources/syslog.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Syslog Acquisition
+
+This module allows crowdsec to expose a syslog server, and ingest logs directly from another syslog server (or any software that knows how to forward logs with syslog).
+
+Only UDP is supported.
+
+Syslog messages must conform either to RFC3164 or RFC5424, and can be up to 2048 bytes long.
```
Comment (buixor) [inline docs/v1.X/docs/user_guide/data_sources/syslog.md:7]: link to rfc

File: pkg/acquisition/modules/syslog/internal/syslogserver.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	udpAddr, err := net.ResolveUDPAddr("udp", fmt.Sprintf("%s:%d", s.listenAddr, s.port))
+	if err != nil {
+		return errors.Wrapf(err, "could not resolve addr %s", s.listenAddr)
+	}
+	udpConn, err := net.ListenUDP("udp", udpAddr)
+	if err != nil {
+		return errors.Wrapf(err, "could not listen on port %d", s.port)
+	}
+	s.udpConn = udpConn
+	err = s.udpConn.SetReadBuffer(1024 * 8) // FIXME probably
```
Comment (buixor) [inline pkg/acquisition/modules/syslog/internal/syslogserver.go:40]: let's make this configurable

**Review by buixor — COMMENTED** [review]
minor coms

**Pull request merged**